pkg/strategy/support/strategy.go

```diff
@@ -77,14 +77,72 @@ func (stop *PercentageTargetStop) GenerateOrders(market types.Market, pos *types
 	return targetOrders
 }
 
-// ResistanceStop is a kind of stop order by detecting resistance
-type ResistanceStop struct {
-	Interval      types.Interval   `json:"interval"`
-	sensitivity   fixedpoint.Value `json:"sensitivity"`
-	MinVolume     fixedpoint.Value `json:"minVolume"`
-	TakerBuyRatio fixedpoint.Value `json:"takerBuyRatio"`
+type TrailingStopTarget struct {
+	TrailingStopCallBackRatio float64 `json:"trailingStopCallBackRatio"`
+	MinimumProfitPercentage float64 `json:"minimumProfitPercentage"`
+}
+
+type TrailingStopControl struct {
+	symbol string
+	market types.Market
+	marginSideEffect types.MarginOrderSideEffectType
+
+	trailingStopCallBackRatio float64
+	minimumProfitPercentage   float64
+
+	CurrentHighestPrice fixedpoint.Value
+	OrderID             uint64
+}
+
+func NewTrailingStopControl(symbol string, market types.Market, marginSideEffect types.MarginOrderSideEffectType, trailingStopCallBackRatio float64, minimumProfitPercentage float64) *TrailingStopControl {
+    var control TrailingStopControl
+
+    control.symbol = symbol
+    control.market = market
+    control.marginSideEffect = marginSideEffect
+
+    control.CurrentHighestPrice = fixedpoint.NewFromInt(0)
+
+    control.trailingStopCallBackRatio = trailingStopCallBackRatio
+    control.minimumProfitPercentage = minimumProfitPercentage
+
+	return &control
 }
 
+func (control *TrailingStopControl) IsHigherThanMin(minTargetPrice float64) bool {
+	targetPrice := control.CurrentHighestPrice.Float64() * (1 - control.trailingStopCallBackRatio)
+
+	return targetPrice >= minTargetPrice
+}
+
+func (control *TrailingStopControl) GenerateTrailingStopOrder(quantity float64) types.SubmitOrder {
```

Comment:
Since you're in the TrailingStopControl, the method name can be just GenerateStopOrder